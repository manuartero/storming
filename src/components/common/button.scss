.button {
  width: 133px;
  height: 34px;

  /* Auto layout */
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 8px 0px;
  flex: none;
  order: 0;
  flex-grow: 0;

  /* visual */
  border: 2px solid #000000;
  border-radius: 16px;
  background: #FFFFFF;
  box-shadow: inset 0px -4px 0px #FFCA3A;
}
