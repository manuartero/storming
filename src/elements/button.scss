@use "/src/styles/colors.scss";

@mixin box-shadow($color) {
  box-shadow: inset -1px -3px 0px $color;
  background: #ffffff; // override
}

.button {
  width: 12vh;
  height: 4vh;

  font-weight: bold;

  border: 2px solid colors.$darkest-grey;
  border-radius: 12px;

  &--disabled {
    border: 2px solid colors.$base-grey;
    box-shadow: none !important;
  }

  &:hover {
    cursor: pointer;
  }

  transition: all 0.2s ease-in-out;
}

.button.player {
  @include box-shadow(colors.$player-primary);
}

.button.enemy1 {
  @include box-shadow(colors.$enemy1-primary);
}

.button.enemy2 {
  @include box-shadow(colors.$enemy2-primary);
}

.button.enemy3 {
  @include box-shadow(colors.$enemy3-primary);
}
