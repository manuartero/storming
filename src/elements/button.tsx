import c from "classnames";

import styles from "./button.module.css";

type Props = {
  player?: PlayerType;
} & React.ComponentProps<"button">;

export function Button({
  player,
  disabled = false,
  className,
  children,
  ...props
}: Props) {
  return (
    <button
      className={c(
        styles.button,
        className,
        disabled && styles.disabled,
        player && styles[player]
      )}
      {...props}
    >
      {children}
    </button>
  );
}
