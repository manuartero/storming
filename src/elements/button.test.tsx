import { render, screen } from "@testing-library/react";
import { Button } from "./button";

describe("<Button />", () => {
  test("renders a button", () => {
    render(<Button>Click Me</Button>);

    const buttonElement = screen.getByRole("button", { name: /click me/i });
    expect(buttonElement).toMatchSnapshot();
  });

  test('applies player class when "player" prop is provided', () => {
    render(<Button player="enemy1">Click Me</Button>);

    const buttonElement = screen.getByRole("button", { name: /click me/i });
    expect(buttonElement).toHaveClass("enemy1");
  });

  test("applies disabled class when disabled prop is true", () => {
    render(<Button disabled>Click Me</Button>);

    const buttonElement = screen.getByRole("button", { name: /click me/i });
    expect(buttonElement).toHaveClass("disabled");
  });
});
