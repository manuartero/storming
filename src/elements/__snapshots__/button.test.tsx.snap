// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Button /> renders a button 1`] = `
<button
  class="button"
>
  Click Me
</button>
`;
